### PR TITLE
[security] Support a list of SSL ECs

### PR DESCRIPTION
**Pull request by rockspore**
Addresses #23235

### CODE REVIEW

**Review by matthewstevenson88 — COMMENTED** [review]
This PR LGTM, but can we add e2e tests to confirm there are no gotchas or other changes needed for these curves?

I'm thinking we can add a couple of tests to `test/cpp/end2end/ssl_credentials_test.cc`. WDYT?

**Comment by rockspore** [comment]
> This PR LGTM, but can we add e2e tests to confirm there are no gotchas or other changes needed for these curves?
> 
> I'm thinking we can add a couple of tests to `test/cpp/end2end/ssl_credentials_test.cc`. WDYT?

I had to create the end2end test in core. Because that's the only way we can currently force TLSv1.2 in SSL credential options. The test was following the h2_ssl_session_reuse_test but I removed the session cache.

File: test/core/end2end/h2_ssl_certs_curves_test.cc [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+  const char* server_key =
+      reinterpret_cast<const char*> GRPC_SLICE_START_PTR(key_slice);
+  grpc_ssl_pem_key_cert_pair pem_cert_key_pair = {server_key, server_cert};
+  auto* cert_config =
+      grpc_ssl_server_certificate_config_create(ca_cert, &pem_cert_key_pair, 1);
+  auto* options = grpc_ssl_server_credentials_create_options_using_config(
+      GRPC_SSL_REQUEST_CLIENT_CERTIFICATE_AND_VERIFY, cert_config);
+  grpc_server_credentials* server_creds =
+      grpc_ssl_server_credentials_create_with_options(options);
+  // This is a hack but we don't have a public API to force TLS version yet.
```
Comment (matthewstevenson88) [inline test/core/end2end/h2_ssl_certs_curves_test.cc:93]: Please explain in the test why we need to force TLS 1.2.

Comment (rockspore) [reply test/core/end2end/h2_ssl_certs_curves_test.cc:93]: Done.

**Review by matthewstevenson88 — APPROVED** [review]

**Comment by mitchallain** [comment]
Any follow up on why this PR was reverted?

**Pull request merged**